### ISSUE TITLE
Inconsistent parameters PolicyStore PersistentStore and Dhcp Enabled

### ISSUE THREAD

**Issue by penenkel** [docker/for-win#2050]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 7C547CCF-E48A-40BB-A5A6-BE5B0880295A/2018-05-18_22-09-35

### Expected behavior
Start docker (for Windows) and begin normal operation.
### Actual behavior
Docker startup fails with:

> Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Inconsistent parameters PolicyStore PersistentStore and Dhcp Enabled

### Information

  - Windows Version:  Windows 10 Pro Version 1803 Build 10.0.17134
  - Docker for Windows Version: 18.05.0-ce-win66 (17760)

As far as I can tell the error is raised in the `New-Switch` function of `MobyLinux.ps1`.
It may be caused by the call to `New-NetIPAddress`, but that is only google-based guesswork on my part.

### Things I tried
- Reset Docker
- Reinstall Docker
- Disable/Enable Hyper-V
- ForceRemoveDocker.ps1
- delete all `Hyper-V Virtual Ethernet Adapter` in device manager

### Things that changed shortly before
- installed newest Windows feature update

--------
The most relevant part of the error message is `Inconsistent parameters PolicyStore PersistentStore and Dhcp Enabled`. Please only add to this issue if you have precisely that error. 

This error has been reported before but the issues either went stale, have no real solution or disappeared miraculously: #486, #720, #408, #1468, #509

The most relevant issue appears to be #2, but that one is quite old and supposedly a fix was integrated way back then.

-----

### Cause
All `Hyper-V Virtual Ethernet Adapters`  displayed `network cable unplugged`, i.e. `disconnected` as status under `Control Panel\All Control Panel Items\Network Connections`.

According to this [blog post](http://www.darrylvanderpeijl.com/inconsistent-parameters-policystore-persistentstore-and-dhcp-enabled/) and other sources, the command `New-NetIPAddress` does not work with disconnected adapters.

### Solution (in my case)
Uninstall `Cisco AnyConnect Secure Mobility Client`

For unknown reasons AnyConnect in combination with the Windows Feature Update blocked the virtual adapters from connecting. Removing AnyConnect solved this problem. I did not test if a reinstallation of AnyConnect would work, as I do not need it at the moment.

**Comment by blockofdynamite** [comment]
Can confirm I'm having the exact same issue.
- Windows Version: Windows 10 Pro Version 1803 Build 10.0.17134.48
- Docker for Windows Version: 18.03.1-ce (9ee9f40)

**Comment by HejVkt** [comment]
I'm having exact. This happened after the system update of Windows, before that everything worked.
crash id 27936D95-418C-4017-B02B-E5054DCDBBA6/2018-05-21_01-10-41

I tried 17.3 ; 17.9 ; 18.03.1-ce(9ee9f40) Docker versions. But it's not working.

**Comment by adamvaul** [comment]
Similar issue here.

Here is the error message I get:

Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: The object already exists.

	   at Docker.Core.Pipe.NamedPipeClient.Send(String action, Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeClient.cs:line 36
	   at Docker.Actions.DoStart(SynchronizationContext syncCtx, Boolean showWelcomeWindow, Boolean executeAfterStartCleanup) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 67
	   at Docker.Actions.<>c__DisplayClass14_0.<Start>b__0() in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 51
	   at Docker.WPF.TaskQueue.<>c__DisplayClass19_0.<.ctor>b__1() in C:\gopath\src\github.com\docker\pinata\win\src\Docker.WPF\TaskQueue.cs:line 59


I am running
![image](https://user-images.githubusercontent.com/1307631/40329553-c47cc554-5cfe-11e8-92db-069acee9e48d.png)

**Comment by penenkel** [comment]
@adamvaul As far as I can tell your error is similar, in that it is caused at the same code segment as mine. Nevertheless it is not the *same* error, so perhaps you should open a separate issue

@jsonGiles IMHO your error bears no resemblance to my problem apart from the obvious effect that Docker doesn't work. 

I recommend that both those comments be deleted/moved to separate issues to keep this one pure and concise. (perhaps also remove any tags that where created due to them)

**Comment by jsonGiles** [comment]
Yes, I agree. Now that I have reread the original issue mine is different. My apologies.

**Comment by penenkel** [comment]
So, this is now an other useless issue report ... the problem disappeared without any action on my part.

It is rather frustrating, after several days of fruitless attempts at solving this (including opening this issue) I gave up and just ignored it for several days. Just now I decided to give it an other try and, what do you know, after an other reset it just works again ... As far as I can tell nothing changed (no new Docker version, no further Windows updates).

**Comment by penenkel** [comment]
and it's back ....

**Comment by penenkel** [comment]
### Cause
All `Hyper-V Virtual Ethernet Adapters`  displayed `network cable unplugged`, i.e. `disconnected` as status under `Control Panel\All Control Panel Items\Network Connections`.

According to this [blog post](http://www.darrylvanderpeijl.com/inconsistent-parameters-policystore-persistentstore-and-dhcp-enabled/) and other sources, the command `New-NetIPAddress` does not work with disconnected adapters.

Perhaps the `MobyLinux.ps1` could be adjusted to query the connection status and produce a more meaningful error message.  Perhaps it should also be mentioned in the docs.

### Solution
Uninstall `Cisco AnyConnect Secure Mobility Client`

For unknown reasons AnyConnect in combination with the Windows Feature Update blocked the virtual adapters from connecting. Removing AnyConnect solved this problem. I did not test if a reinstallation of AnyConnect would work, as I do not need it at the moment.

**Comment by necroTaxonomist** [comment]
Does anybody have a solution to this yet that doesn't require uninstalling `Cisco AnyConnect Secure Mobility Client`?

I'm on a work computer and I'd prefer not to need to just uninstall things.

**Comment by necroTaxonomist** [comment]
Honestly, everything was working fine for me until yesterday. Didn't even install any sort of update. Just turned it on one day and started getting this error.

**Comment by anobleperson** [comment]
Similar to @necroTaxonomist  DockerCE was working fine for me and then stopped.  Following steps worked for me:
1. Uninstal DockerCE
2. Disable CiscoAnyConnect from services
3. Re-instal DockerCE
4. Restart CiscoAnyConnect from services

The uninstall may not be necessary, but it's the steps I took after several hours of frustration.  I've got CiscoAnyConnect 4.4.02039

**Comment by mbouxsein** [comment]
#/remove-lifecycle stale

**Comment by mbouxsein** [comment]
#/remove-lifecycle stale

**Comment by monroecheeseman** [comment]
I'm a new user that has Cisco AnyConnect Secure Mobility Client Version 4.5.04029 installed due to a work policy.  I installed the latest stable version of Docker CE for Windows, received the same error "Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Inconsistent parameters PolicyStore PersistentStore and Dhcp Enabled".  Rebooted same error.  Crash log is C5BB6FA2-F5BC-4939-892E-926CBC9FDD44/20181114075528.  Unfortunately I am unable to uninstall AnyConnect.

**Comment by maxnbk** [comment]
I have the same issue, but restarting the AnyConnect service re-kills any networking, and stopping it restores it. Unfortunately, I do need to use both of these at the same time.

**Comment by yarix** [comment]
Same here:
win 10, 1803
Cisco any connect 4.5.04029

docker won't start due to _Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Inconsistent parameters PolicyStore PersistentStore and Dhcp Enabled_

workaround:
stop Cisco agent, restart docker, start Cisco agent.

**Comment by NerdyItalian** [comment]
Had this issue today - the work around mentioned by @anobleperson worked for me.

**Comment by jonathanmilley** [comment]
For my other NAT switches in Hyper-V I can fix the no connection issue by going into Network connection properties for the particular switch, disable "Cisco AnyConnect Network Access Manager Filter Driver" under "This connection uses the following items: " however I can't manage to do that for docker as it seems to delete the connection when I hit the restart button.

**Comment by jonathanmilley** [comment]
just after typing that, I got it fixed by doing the following:
In Hyper-V virtual switch manager I manually created a new switch called DockerNAT, then changed its settings, then restarted docker. now I'm back up and running.

**Comment by ardikars** [comment]
I got same issue, how to disable Cisco AnyConnect? I can't find in Services.

**Comment by monroecheeseman** [comment]
> I got same issue, how to disable Cisco AnyConnect? I can't find in Services.

It could be you don't have Cisco Anyconnect? (It's installed by the network team on my corporate laptop)  Mine shows up as a world icon in the system tray.  
https://www.cisco.com/c/en/us/products/security/anyconnect-secure-mobility-client/index.html

**Comment by ardikars** [comment]
> > I got same issue, how to disable Cisco AnyConnect? I can't find in Services.
> 
> It could be you don't have Cisco Anyconnect? (It's installed by the network team on my corporate laptop) Mine shows up as a world icon in the system tray.
> https://www.cisco.com/c/en/us/products/security/anyconnect-secure-mobility-client/index.html

Mine fixed after restarting my laptop.

**Comment by andremmfaria** [comment]
Stopping Cisco AnyConnect services, reinstalling docker and starting them soved it for me.

These are the services:

![image](https://user-images.githubusercontent.com/400238/54560161-75b67e80-49a0-11e9-9690-9fcb5a336bf6.png)

Thanks :D

**Comment by ack72tdp** [comment]
Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Inconsistent parameters PolicyStore PersistentStore and Dhcp Enabled

**Comment by ack72tdp** [comment]
docker fails to start .. with this message ... Any ideas?

Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Inconsistent parameters PolicyStore PersistentStore and Dhcp Enabled


   at Docker.Core.Pipe.NamedPipeClient.Send(String action, Object[] parameters) in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeClient.cs:line 36
   at Docker.Actions.DoStart(SynchronizationContext syncCtx, Boolean showWelcomeWindow, Boolean executeAfterStartCleanup) in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 92
   at Docker.Actions.<>c__DisplayClass21_0.<ResetToDefault>b__0() in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 155
   at Docker.WPF.TaskQueue.<>c__DisplayClass19_0.<.ctor>b__1() in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.WPF\TaskQueue.cs:line 59

**Comment by alexander-sk** [comment]
My solution:
go to "vEthernet(Default Switch)" and "vEthernet(DockerNAT)" adapters properties and uncheck "Cisco AnyConnect Network Access Manager Filter Driver".
Docker runs from administartor.
AnyConnect version: 4.5.02036
Docker version: 2.0.0.3(31259)

**Comment by ack72tdp** [comment]
Perfect .. that works great ...Thanks a lot. Appreciate your help ..

**Comment by ack72tdp** [comment]
I run into another issue now ..

C:\Users\chadaa>docker run hello-world
docker: Error response from daemon: CreateComputeSystem 7a107d2ac574c296aa2a91b3c5c7c7d4df130c880e5561616459816320fa0ed8: Logon failure: the user has not been granted the requested logon type at this computer.
(extra info: {"SystemType":"Container","Name":"7a107d2ac574c296aa2a91b3c5c7c7d4df130c880e5561616459816320fa0ed8","Owner":"docker","IgnoreFlushesDuringBoot":true,"LayerFolderPath":"C:\\ProgramData\\Docker\\windowsfilter\\7a107d2ac574c296aa2a91b3c5c7c7d4df130c880e5561616459816320fa0ed8","Layers":[{"ID":"096ffcfb-0053-52e8-a684-d909564e3bc7","Path":"C:\\ProgramData\\Docker\\windowsfilter\\be40aba52549de5ca1a020012edd2a6dd985ee0802fc9c988ad7094b4f878f9c"},{"ID":"cdfee02e-92de-567e-bf3e-37f66e1760ca","Path":"C:\\ProgramData\\Docker\\windowsfilter\\b87d9b18ee7fbd97be118149c938c86fab0095afdc50819d372c594aae2a5636"},{"ID":"cc596128-a4f0-5b8b-891f-52b8e7c172e6","Path":"C:\\ProgramData\\Docker\\windowsfilter\\92a7e670c046459321c9e9f5f1cff4e5f44a1e6a35c49d163d25bb1f87bcf257"},{"ID":"bc280e00-ef8f-5a0d-b330-6255e43203cb","Path":"C:\\ProgramData\\Docker\\windowsfilter\\19ff3f35ef63721f70d8f21c0ff8e42660a8f68db98e937dde1748f83706a0e5"}],"HostName":"7a107d2ac574","HvPartition":true,"EndpointList":["3c563436-df55-4375-abc2-f005d58bdc46"],"HvRuntime":{"ImagePath":"C:\\ProgramData\\Docker\\windowsfilter\\92a7e670c046459321c9e9f5f1cff4e5f44a1e6a35c49d163d25bb1f87bcf257\\UtilityVM"},"AllowUnqualifiedDNSQuery":true}).


Any ideas?

Thanks

**Comment by SteveALee** [comment]
Just for info I get same error but while  do not have Cisco Anywhere connect (tha tI can find) I do use Cisco WebX. have been using it for ages but not noticed problems  - perhaps they updated something

**Comment by cbrsa** [comment]
My fix...  Anyconnect -> Network -> Mange Networks -> VPN -> check Allow local (LAN) access

**Comment by praveenvelampudi7** [comment]
I have a similar issue:

Once I install and restart my computer and dockers begins to start up I run into the following error:

Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Inconsistent parameters PolicyStore PersistentStore and Dhcp Enabled


   at Docker.Core.Pipe.NamedPipeClient.Send(String action, Object[] parameters) in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeClient.cs:line 36
   at Docker.Actions.DoStart(SynchronizationContext syncCtx, Boolean showWelcomeWindow, Boolean executeAfterStartCleanup) in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 92
   at Docker.Actions.<>c__DisplayClass21_0.<ResetToDefault>b__0() in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 155
   at Docker.WPF.TaskQueue.<>c__DisplayClass19_0.<.ctor>b__1() in C:\workspaces\stable-18.09.x\src\github.com\docker\pinata\win\src\Docker.WPF\TaskQueue.cs:line 59

**Comment by verintjgray** [comment]
Same. I  uploaded a crash report, id: 6E2AC365-62E7-41F3-AC22-18C0D80B830D/20190708165618

**Comment by verintjgray** [comment]
Good morning. Is anyone looking into this issue? If the issue is compatibility with Cisco AnyConnect, can Docker work on trying to craft a workaround, or even better, solve the incompatibility? A lot of corporate users are required to be on Windows 10 with AnyConnect running and are not allowed to disable or remove it. I just tried to get it to work again with Edge channel release but still no luck. Uploaded crash report: 0A9AC16D-A03A-45DA-94B9-482917C72AAD/20190725135338

**Comment by gandolfmania** [comment]
I am new to docker and faced the same issue while installing docker for windows on my office laptop with Windows 10 pro installed with Cisco AnyConnect.

My observations:

1. Docker network adapter "vEthernet(DockerNAT)" appears as Untrusted network adapter in the "Control Panel->Network Connections" section.
2. And in Cisco AnyConnect connections from untrusted servers are blocked by default.

The workaround I tried is as follows:

- Open Cisco AnyConnect Secure Mobile Client application
- Go to "VPN-> Preferences" and uncheck the "Block connections to untrsuted servers" check box.
- Restart "Docker Desktop" application.
And BINGO Docker Desktop will be up and running.

### NOTE:
**Whenever Docker Desktop is not in use please do not forget to check back the "Block connections to untrsuted servers" check box in "Cisco AnyConnect Secure Mobile Client application" to avoid any unwanted hacks**

**Comment by roi972** [comment]
I've also encountered the same problem.
As I don't have access to Cisco AnyConnect settings I 'fixed' this by doing this (both I got from this thread):

1. Disable "Cisco AnyConnect Network Access Manager Filter Driver" on the vEthernet(Default Switch) - in the properties.
2. Create the DockerNAT adapter via Hyper-V settings with 'private' setting.

After that, restart Docker as Administrator and it works.

**Comment by ctaggart** [comment]
My corporate laptop was just reimaged with Windows 10 1803 and and comes with Cisco AnyConnect. I hit this error with the latest stable Docker 2.1.0.1, so you can add that label to the list on this issue. I remember doing a workaround like this about a year ago to make it work.

The diagnostics ID is:
F605830E-86C9-46D4-BA32-836E4A1A6506/20190901103551

I'm going to attempt allowing untrusted servers first:
![image](https://user-images.githubusercontent.com/80104/64075750-283d7d80-cc71-11e9-8ea5-cc0cd1b4da93.png)
